architecture/standards/0007-java-pre-requisite.md

```diff
@@ -0,0 +1,53 @@
+# ADR-0007 - Gradle and Java as a pre-requisite
+
+## Date
+
+2024-06-26
+
+## Context
+
+There have been discussions on embedding a Java runtime in the Gradle distribution.
+This would allow users to run Gradle without having to install a Java runtime beforehand.
+Which is something that can ease Gradle adoption outside of the JVM ecosystem.
+
+Gradle has different runtime components:
```

Comment:
💭 Maybe this section can be restructured to be more clear about there being 3 different ways to invoke Gradle (the wrapper, the tooling API, or running the `gradle` launcher directly), each of which will result in 3 different types of component processes spawned from the Gradle distribution.

Reply:
Added something